frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/JoinTablePicker/JoinTablePicker.tsx

```diff
@@ -25,80 +28,90 @@ interface JoinTablePickerProps {
   tableName: string | undefined;
   color: string;
   isReadOnly: boolean;
-  isModelDataSource: boolean;
   columnPicker: ReactNode;
   onChange?: (table: Lib.Joinable) => void;
 }
 
 export function JoinTablePicker({
   query,
-  table,
+  table: joinable,
   tableName,
   color,
   isReadOnly,
-  isModelDataSource,
   columnPicker,
   onChange,
 }: JoinTablePickerProps) {
-  const metadata = useSelector(getMetadata);
   const dispatch = useDispatch();
+  const onChangeRef = useLatest(onChange);
+  const queryRef = useLatest(query);
 
-  const databaseId = useMemo(() => {
-    return Lib.databaseID(query);
-  }, [query]);
-
-  const databases = useMemo(() => {
-    const database = metadata.database(databaseId);
-    return [database, metadata.savedQuestionsDatabase()].filter(Boolean);
-  }, [databaseId, metadata]);
+  const [isDataPickerOpen, setIsDataPickerOpen] = useState(!joinable);
+  const databaseId = useMemo(() => Lib.databaseID(query), [query]);
 
   const pickerInfo = useMemo(() => {
-    return table ? Lib.pickerInfo(query, table) : null;
-  }, [query, table]);
+    return joinable ? Lib.pickerInfo(query, joinable) : null;
+  }, [query, joinable]);
 
   const tableId = pickerInfo?.tableId ?? pickerInfo?.cardId;
-  const tableFilter = (table: Table) => !tableId || table.db_id === databaseId;
+  const sourceCardId = pickerInfo?.cardId;
+  const { data: sourceCard } = useGetCardQuery(
+    sourceCardId ? { id: sourceCardId } : skipToken,
```

Comment:
This seems wrong. `joinable` is already a card. We need to fix initialization to be able to use it directly

Reply:
4c9fc81336a03897abc9b1f060970231f38ce46c + 721568b5634cfc932af273126f67b98a5f17eddb + 8a58579328bd90f7102a5204233b14e7f816c770